tests/disk/disk.go

```diff
@@ -82,28 +77,30 @@ func (disk *Disk) createDiskImage() error {
 	return nil
 }
 
-// CreateLoopDevice creates a loopback device if the disk is not present
-func (disk *Disk) CreateLoopDevice() error {
-	if disk.DiskPath == "" {
-		var err error
-		if _, err = os.Stat(disk.imageName); err != nil && os.IsNotExist(err) {
-			err = disk.createDiskImage()
-			if err != nil {
-				return err
-			}
-		} else if err != nil {
-			return err
-		}
-		deviceName := getLoopDevName()
-		devicePath := "/dev/" + deviceName
-		// create the loop device using losetup
-		createLoopDeviceCommand := "losetup " + devicePath + " " + disk.imageName + " --show"
-		err = RunCommandWithSudo(createLoopDeviceCommand)
-		if err != nil {
-			return fmt.Errorf("error creating loop device. Error : %v", err)
-		}
-		disk.DiskPath = devicePath
+// CreateLoopDevice creates a loop device if the disk is not present
+func (disk *Disk) CreateLoopDevice(imgDir string) error {
+	if len(disk.DiskPath) != 0 {
+		return fmt.Errorf("aborting disk creation. " +
+			"Error: disk path is already set on Disk obj")
+	}
+
+	err := disk.createDiskImage(imgDir)
+	if err != nil {
+		return err
+	}
+
+	// create the loop device using losetup
+	createLoopDeviceCommand := "losetup -P " + "/dev/" + getLoopDevName() + " " + disk.imageName + " --show"
+	var devicePath []byte
+	devicePath, err = RunCommand(createLoopDeviceCommand)
+	if err != nil {
+		return fmt.Errorf("error creating loop device. Error : %v", err)
 	}
+	// Trim trailing newline
+	devicePath = devicePath[:len(devicePath)-1]
+
+	disk.DiskPath = string(devicePath)
```

Comment:
Why is this conversion needed here? Its already a string.

Reply:
It's a byte slice